src/Monotonic.cpp

```diff
@@ -209,10 +332,16 @@ class MonotonicVisitor : public IRVisitor {
 
     void visit_lt(const Expr &a, const Expr &b) {
         a.accept(this);
-        Monotonic ra = result;
+        ConstantInterval ra = result;
         b.accept(this);
-        Monotonic rb = result;
-        result = unify(flip(ra), rb);
+        ConstantInterval rb = result;
+        result = unify(negate(ra), rb);
+        if (result.has_lower_bound()) {
+            result.min = std::max<int64_t>(result.min, -1);
```

Comment:
Comments here would be good. E.g. "It's possible for the value to flip from true to false, which is a step of -1, because ..."